relay-server/src/actors/project.rs

```diff
@@ -699,7 +726,7 @@ impl Project {
     /// request's scoping. Otherwise, this function returns partial scoping from the `request_meta`.
     /// See [`RequestMeta::get_partial_scoping`] for more information.
     fn scope_request(&self, meta: &RequestMeta) -> Scoping {
-        match self.state() {
+        match self.get_valid_state() {
```

Comment:
This now clones the `Arc` around the state instead of just taking its reference, just to keep the code simple. The alternative would be to offer `get_valid_state_clone()` and `get_expiry_state_clone()` methods as well.